libs/utils/src/lib.rs

```diff
@@ -14,11 +14,9 @@ pub mod simple_rcu;
 /// append only ordered map implemented with a Vec
 pub mod vec_map;
 
-// Async version of SeqWait. Currently unused.
-// pub mod seqwait_async;
```

Comment:
Right, this PR changes `seqwait.rs` to be async, and removes the old unused `seqwait_async.rs` variant. The implementation in  `seqwait_async.rs` was slightly different, it used a BTreeMap internally instead of a BinaryHeap. I didn't see any particular reason for those differences, so it seems best to stick to the implementation we had, and just change it to use a tokio channel instead of the std library one.